### PR TITLE
Add a bit about crater

### LINKED ISSUE #10: "Using the compiler testing framework"
This section should describe how to compiler is tested. Substantial work has been done but there remains more to document:

- [x] discuss bors, travis, platform emulation (#66 )
- [ ] crater: what is it and why is it useful? How do you request a crater run and when should you to do so? (#151)
- [ ] maybe talk about "perf runs" too (#151)
- [x] rustdoc tests (cc @QuietMisdreavus)
- [x] talk about unit tests from libs and things (#62)
- [x] Doctests: How to run the doc tests in the standard library.  Other doc testing steps like linkchecker. (#62)
- [x] distcheck (#62)
- [x] mention that tests also get run for individual packages, such as bootstrap, cargo, rls, rustfmt, miri, clippy, cargotest. (#62)

### PR DESCRIPTION
**Pull request by mark-i-m**
closes #10

r? @Mark-Simulacrum

### CODE REVIEW

File: src/tests/intro.md [hunk 1]
```diff
@@ -170,7 +170,38 @@ communicate with the server to coordinate running tests (see
 
 ## Crater
 
-TODO
+[Crater](https://github.com/rust-lang-nursery/crater) is a tool for compiling
+and running tests for _every_ crate on [crates.io](https://crates.io/). It is
```
Comment (Mark-Simulacrum) [inline src/tests/intro.md]: and also many crates from GitHub... I'm not exactly sure what the search parameters are, but I know we at least *try*.

File: src/tests/intro.md [hunk 2]
```diff
@@ -170,7 +170,38 @@ communicate with the server to coordinate running tests (see
...
+mainly used for checking for extent of breakage when implementing potentially
+breaking changes.
```
Comment (Mark-Simulacrum) [inline src/tests/intro.md]: and ensuring lack of breakage by running beta vs. stable "builds."

File: src/tests/intro.md [hunk 3]
```diff
@@ -170,7 +170,38 @@ communicate with the server to coordinate running tests (see
...
+You should request a crater run if your PR makes large changes to the compiler
+or could cause breakage. If you are unsure, feel free to ask your PR's reviewer.
+
+### Requesting Crater Runs
+
+The rust team maintains a few machines that can be used for running crater runs
+on the changes introduced by a PR. If your PR needs a crater run, leave a
+comment for the triage team in the PR thread. Your will be enqueued by the
+triage team and the results will be posted when they are ready. A crater run
+usually takes a few days (as of this writing).
```
Comment (Mark-Simulacrum) [inline src/tests/intro.md]: Please inform the team whether you require a "check-only" crater run, a "build only" crater run, or a "build-and-test" crater run. The difference is primarily in time; the conservative (if you're not sure) option is to go for the build-and-test run. If making changes that will only have an effect at compile-time (e.g., implementing a new trait) then you only need a check run.

Check runs will take around ~3-4 days, with the other two taking 5-6 days on average.

File: src/tests/intro.md [hunk 4]
```diff
@@ -170,7 +170,38 @@ communicate with the server to coordinate running tests (see
...
+
+While crater is really useful, it is also important to be aware of a few caveats:
+
+- Not all code is on crates.io! There is a lot of code in repos on GitHub and
+  elsewhere. Also, companies may not wish to publish their code. Thus, a
+  successful crater run is not a magically green light that there will be no
+  breakage; you still need to be careful.
+
+- Crater only runs Linux builds (on x86_64, I believe). Thus, other
```
Comment (Mark-Simulacrum) [inline src/tests/intro.md]: Yes, we only test / compile for x86_64-unknown-linux-gnu.

**Comment by mark-i-m** [comment]
@Mark-Simulacrum Thanks for the feedback! I basically copy/pasted what you wrote :)

Also I added 
- that `@bors try` needs to pass (that's true, right?).
- a little about perf runs

File: src/tests/intro.md [hunk 5]
```diff
@@ -170,7 +170,63 @@ communicate with the server to coordinate running tests (see
...
+
+- Crater only runs Linux builds on x86_64. Thus, other architectures and
+  platforms are not tested. Critically, this includes Windows.
+
+- Many crates are not tested. This could be for a lot of reasons, including
+  that the crate doesn't compile any more (e.g. used old nightly features),
+  has broken or flaky tests, requires network access, or other reasons.
+
+- Before crater can be run, `@bors try` needs to suceed. This means that if
+  your code doesn't compile or fails tests, you cannot run crater.
```
Comment (Mark-Simulacrum) [inline src/tests/intro.md]: Failing tests is currently acceptable (we don't run them); we just need the code to compile on x86_64-unknown-linux-gnu.

File: src/tests/intro.md [hunk 6]
```diff
@@ -170,7 +170,63 @@ communicate with the server to coordinate running tests (see
...
+  breakage; you still need to be careful.
+
+- Crater only runs Linux builds on x86_64. Thus, other architectures and
+  platforms are not tested. Critically, this includes Windows.
+
+- Many crates are not tested. This could be for a lot of reasons, including
+  that the crate doesn't compile any more (e.g. used old nightly features),
+  has broken or flaky tests, requires network access, or other reasons.
+
+- Before crater can be run, `@bors try` needs to suceed in building artifacts.
```
Comment (Mark-Simulacrum) [inline src/tests/intro.md]: s/suceed/succeed

Comment (mark-i-m) [reply src/tests/intro.md]: fixed

**Comment by mark-i-m** [comment]
Thanks!

**Pull request merged**